Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -2006,6 +2013,60 @@ void MatrixWorkspace::updateCachedDetectorGrouping(const size_t index) const {
   setDetectorGrouping(index, getSpectrum(index).getDetectorIDs());
 }
 
+void MatrixWorkspace::buildDefaultSpectrumDefinitions() {
+  const auto &detInfo = detectorInfo();
+  size_t numberOfDetectors{detInfo.size()};
+  size_t numberOfSpectra{0};
+  if (detInfo.isScanning()) {
+    for (size_t i = 0; i < numberOfDetectors; ++i)
+      numberOfSpectra += detInfo.scanCount(i);
```

Comment:
I was initially thinking that this behaviour could like in `DetectorInfo` but I nowthink having it in the client code is actually correct as you have done.